app/controllers/users/registrations_controller.rb

```diff
@@ -0,0 +1,75 @@
+class Users::RegistrationsController < Devise::RegistrationsController
+  # before_action :configure_sign_up_params, only: [:create]
+  # before_action :configure_account_update_params, only: [:update]
+
+  def update
+    # Call the parent implementation to update the user's information
+    super
+
+    # Handle the avatar separately, if it was changed
+    return unless params[:user][:avatar].present?
+
+    current_user.avatar.attach(params[:user][:avatar])
+  end
+
+  # # GET /resource/sign_up
+  # def new
+  #   super
+  #   if params[:avatar].present?
+  #     current_user.avatar.attach(params[:avatar])
+  #   end
+  # end
+
+  # POST /resource
+  def create
+    super do |resource|
+      resource.avatar.attach(params[:user][:avatar]) if params[:user][:avatar].present?
+    end
+  end
+
+  # GET /resource/edit
+  # def edit
+  #   super
+  # end
+
+  # PUT /resource
+  # def update
+  #   super
+  # end
+
+  # DELETE /resource
+  # def destroy
```

Comment:
### Optional
- [ ] Kindly delete commented piece of code, it makes our code base looks messy and difficult to understand